src/modules/app/containers/Switcher.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+
+import * as React from 'react'
+import { connect } from 'react-redux'
+import LandingPage from '../../../routes/landing/containers/LandingPage'
+import Spinner from 'react-spinkit'
+import Layout from '../../layout/components/Layout'
+import LocationLayout from '../../layout/containers/LocationLayout'
+import MainDisclaimerPage from '../../../routes/main-disclaimer/components/MainDisclaimerPage'
+import GeneralFooter from '../../layout/components/GeneralFooter'
+import GeneralHeader from '../../layout/components/GeneralHeader'
+import CategoriesPage from '../../../routes/categories/containers/CategoriesPage'
+import EventsPage from '../../../routes/events/containers/EventsPage'
+import ExtrasPage from '../../../routes/extras/containers/ExtrasPage'
+import DisclaimerPage from '../../../routes/disclaimer/containers/DisclaimerPage'
+import SearchPage from '../../../routes/search/containers/SearchPage'
+import PdfFetcherPage from '../../../routes/pdf-fetcher/containers/PdfFetcherPage'
+import { LANDING_ROUTE } from '../routes/landing'
+import { MAIN_DISCLAIMER_ROUTE } from '../routes/mainDisclaimer'
+import { CATEGORIES_ROUTE } from '../routes/categories'
+import { EVENTS_ROUTE } from '../routes/events'
+import { EXTRAS_ROUTE } from '../routes/extras'
+import { DISCLAIMER_ROUTE } from '../routes/disclaimer'
+import { SEARCH_ROUTE } from '../routes/search'
+import { PDF_FETCHER_ROUTE } from '../routes/pdfFetcher'
+
+import Payload from '../../endpoint/Payload'
+import Failure from '../../common/components/Failure'
+
+type Props = {
+  viewportSmall: boolean,
+  currentRoute: string,
+  citiesPayload: Payload,
+  categoriesPayload: Payload,
+  eventsPayload: Payload,
+  extrasPayload: Payload,
+  disclaimerPayload: Payload
+}
+
+/**
+ * Renders different Pages depending on the current route. If the needed data is not available, a LoadingSpinner is rendered
+ */
+class Switcher extends React.Component<Props> {
+  static getDisplayComponent (page: React.Node, payload: Payload): React.Node {
+    const LoadingSpinner = () => <Spinner name='line-scale-party' />
+    if (payload.isFetching) {
+      return <LoadingSpinner />
+    } else if (payload.data) {
+      return page
+    } else if (payload.error) {
+      return <Failure error={payload.error} />
+    } else {
+      return null
+    }
+  }
+
+  getPage (): React.Node {
```

Comment:
this should be called renderPage() because executing this method is only valid inside of an render() function.

Reply:
done